src/Chapter3.hs

```diff
@@ -971,6 +1148,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Enum, Bounded, Eq, Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend x = case x of
+    Saturday -> True
+    Sunday -> True
+    _ -> False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay x = case x of
+    Sunday -> Monday
+    _ -> succ x
+
+daysToParty :: DayOfWeek -> Int
+daysToParty x = case x of
+    Saturday -> 6
+    Sunday -> 5
+    _ -> length (enumFromTo x Friday) - 1
```

Comment:
That is right that you need to use the instance of `Enum` here. Note that is you use `mod` with the number of days in the week you won't need to have any special checks 🙂